core/job/spec.go

```diff
@@ -162,12 +165,39 @@ func (s Specs) ToFullNameAndSpecMap(projectName tenant.ProjectName) map[FullName
 	return fullnameAndSpecMap
 }
 
+func (s Specs) Validate() ([]*Spec, error) {
```

Comment:
If we want to use validation method, let's change method `Validate` to only return one result, which is error (if encountered error or not). This way it is clear that what this method does it validation.

In case we need a unique list of job, we can have another method. Maybe name it `GetUnique` or something. The main task for this method is to just get the unique results only.

Agreed that this operation will require two times of loop. However, the `Specs` consists of pointers.